File: 2000 98 & ME/QuickStart/howto/samples/Remoting/hello/VB/object.vb
imports System
imports System.Runtime.Remoting
imports System.Runtime.Remoting.Channels.TCP

namespace RemotingSamples

  public class HelloServer 
    inherits MarshalByRefObject
  
    public sub HelloServer()
      Console.WriteLine("HelloServer activated")
    end sub

    public function HelloMethod(name as String) as String
      Console.WriteLine("Hello.HelloMethod : {0}", name)
      HelloMethod = "Hi there " + name
    end function
    
  end class
  
end namespace

File: 2000 98 & ME/QuickStart/howto/samples/Remoting/hello/object.cs
using System;
using System.Runtime.Remoting;
using System.Runtime.Remoting.Channels.TCP;

namespace RemotingSamples {
  public class HelloServer : MarshalByRefObject {
  
    public HelloServer()
    {
      Console.WriteLine("HelloServer activated");
    }
    
    
    ~HelloServer()
    {
      Console.WriteLine("Object Destroyed");
    }  
    
    public String HelloMethod(String name) 
    {
      Console.WriteLine("Hello.HelloMethod : {0}", name);                        
      return "Hi there " + name;
    }
  }
}    
